enzyme/Enzyme/CApi.h

```diff
@@ -117,13 +117,42 @@ typedef enum {
   DEM_ForwardModeSplit = 4,
 } CDerivativeMode;
 
+/// Generates a new function based on the input, which uses forward-mode AD.
+///
```

Comment:
We also need to document the formal function: https://enzyme.mit.edu/doxygen/classEnzymeLogic.html#a8e6ba5c54950f1ebcdfcbd7ce2a16083

I'm slightly inclined to have the C API documentation be light, and instead forward one to see the function which it wraps.